### DISCUSSION TITLE
Problem closing connection

### DISCUSSION THREAD

**Discussion by ingdesarr1** [FreeOpcUa/opcua-asyncio#1637] [Q&A]
Good afternoon. I'm working with asyncua, using sync to my proccess, and I have a problem closing the connection. I made my connection, send the values to my variables and everything works fine. When I going to connect again, I check the connection using the function check_connection(), but if I close the connection and check again appears this error:

could not post <coroutine object Client.check_connection at 0x000002846B89EEC0> since asyncio loop in thread has not been started or has been stopped

and can't connect me again. How can I solve this? Thanks

**Comment by schroeder-** [comment]
`check_connection` only works if you are still connect, after you call disconnect, this function is not working anymore. The function is only to check if connection to the server is still alive.

**Reply by ingdesarr1** [reply]
Yes, I understand, but when I close the connection, this error appears. How can I handle it?

could not post <coroutine object Client.check_connection at 0x000002846B89EEC0> since asyncio loop in thread has not been started or has been stopped
